practitioner/src/main/java/org/smartregister/extension/rest/PractitionerDetailsResourceProvider.java

```diff
@@ -431,9 +431,11 @@ private List<String> getLocationIdentifiersByOrganizations(List<Organization> or
                     organizationAffiliationObj = (OrganizationAffiliation) organizationAffiliation;
                     List<Reference> locationList = organizationAffiliationObj.getLocation();
                     for (Reference location : locationList) {
-                        if (location != null && location.getReference() != null && locationsIdentifiers.size() > 0 &&
-                                !(locationsIdentifiers.contains(location.getReference()))) {
-                                locationsIdentifiers.add(location.getReference());
+                        if (location != null
+                                && location.getReference() != null
+                                && locationsIdentifiers != null
+                                && !(locationsIdentifiers.contains(location.getReference()))) {
+                            locationsIdentifiers.add(location.getReference());
```

Comment:
For this one if we don't need to maintain the insertion order I think we can use a Set instead to for a performance gain. `list.contains` has a **O(n)** while `set.add` has **O(1)**

Reply:
Right, makes sense. 
Let me update it to use `Set` data structure.